src/Compilers/CSharp/Portable/Lowering/Instrumentation/LocalStateTracingInstrumenter.cs

```diff
@@ -0,0 +1,621 @@
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreUInt16,
+                    SpecialType.System_Int32 or SpecialType.System_UInt32
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreUInt32,
+                    SpecialType.System_Int64 or SpecialType.System_UInt64
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreUInt64,
+                    SpecialType.System_Single
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreSingle,
+                    SpecialType.System_Double
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreDouble,
+                    SpecialType.System_Decimal
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreDecimal,
+                    SpecialType.System_String
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreString,
+                    _ when variableType.IsPointerOrFunctionPointer()
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStorePointer,
+                    _ when !variableType.IsManagedTypeNoUseSiteDiagnostics
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreUnmanaged,
+                    _ when variableType.IsRefLikeType && !hasOverriddenToString(variableType)
+                        => null, // not possible to invoke ToString on ref struct that doesn't override it
+                    _ when variableType.TypeKind is TypeKind.Struct
+                        // we'll emit ToString constrained virtcall to avoid boxing the struct
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreString,
+                    _
+                        => WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreObject,
+                }
+            };
+
+            static bool hasOverriddenToString(TypeSymbol variableType)
+                => variableType.GetMembers("ToString").Any(m => m.GetOverriddenMember() is not null);
+
+            if (!overloadOpt.HasValue)
+            {
+                return null;
+            }
+
+            // LogLocalStoreLocalAlias does not have a corresponding parameter version,
+            // since it is not possible to assign address of a local to a by-ref parameter.
+            Debug.Assert(enumDelta == 0 || overloadOpt.Value != WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__LogLocalStoreLocalAlias);
+
+            var overload = overloadOpt.Value + enumDelta;
+
+            var symbol = GetWellKnownMethodSymbol(overload, syntax);
+            if (symbol is not null)
```

Comment:
>if (symbol is not null) [](http://example.com/codeflow?start=12&length=23)

When will the well-known members be missing, other than in error situations?

Reply:
I guess we can require all the specialized overloads to be present.